### DISCUSSION TITLE
Anyone have a Powershell highlighter?

### DISCUSSION THREAD

**Discussion by vincentparrett** [pyscripter/SynEdit#45] [Q&A] [answered]
I haved one created many years ago with SynGen but have somehow managed to lose the definition file - it's never been a great highlighter anyway.

**Comment by pyscripter** [comment]
I am afraid not, but it would be great to have one.

**Reply by pyscripter** [reply]
I am not sure whether it helps but I found a powershell highlighter for the Lazarus Synedit in https://pinote.sourceforge.io/page1.html.
Attached are the relevant files.
[mysehighlighterpowershell.zip](https://github.com/pyscripter/SynEdit/files/10087518/mysehighlighterpowershell.zip)

**Reply by vincentparrett** [reply]
Opened it, saw the GPL license, closed it.

**Comment by pyscripter** [comment]
@JaFi-cz
You have a Powershell highlighter in PSPad.  Would you like to contribute it to SynEdit?

**Comment by JaFi-cz** [comment, marked as answer]
I have PowerShell as syntax definition for General highlighter. My SynGeneralSyn is highly modified in compare to standard one.
I can contribute both, but you need to check it. is it OK for you? Now I am again stuck in job work without more free time.
![User Highlighter definition]

(https://user-images.githubusercontent.com/83579818/203911786-13c555b8-07f1-4deb-af55-20a5a046fd34.png)

**Reply by pyscripter** [reply]
Looks good!  Could you please upload here both you SynGeneralSyn as well as the Powershell highlighter.   We will then can better assess them.

**Comment by JaFi-cz** [comment]
Because I don't think my SynGeneralSyn is compatible, I will send it this way. Take a look, if you need anything, ask me.
Package contains General highlighter + powershell defnition. I used external INI files. There is common settings section with comments setting. Important is KeywordChars variable, what says highlighter what to include into KeywordChars

[SynHighlighterGeneral.zip](https://github.com/pyscripter/SynEdit/files/10095943/SynHighlighterGeneral.zip)

**Reply by pyscripter** [reply]
Your SynHighlighterGeneral looks quite powerful.  It even supports code foldling.  
However if we decide to replace the existing one with your version, It would be quite useful to have a design-time editor similar to the form you have shown above and the ability to load values from ini files at design time.   So then, one would not have to distribute ini files.

**Comment by JaFi-cz** [comment]
I will try to cut it off and make the editor standalone + provide functions
for load syntax file

Jan Fiala

Dne ne 27. 11. 2022 21:48 uživatel pyscripter ***@***.***>
napsal:

**Reply by pyscripter** [reply]
@JaFi-cz 
Hi Jan,

Any news on this?  I would like to integrate your General highlighter to this repo.  Is that OK?   Would you like to also contribute:
- The code that saves/loads to Ini files?
- The Ini editor that could serve as a property editor?
- Your collection of INI based highlighters?

This came back to my mind in the context of this [discussion](https://en.delphipraxis.net/topic/12417-synedit-now-has-annotated-scrollbars/).

Thanks!

**Comment by pyscripter** [comment]
Jan's ini based highlighter is now in the code base. See https://github.com/pyscripter/SynEdit/discussions/138.  A Powershell highlighter is now included.

**Reply by JaFi-cz** [reply]
Thank you.
